semgrep/tests/e2e/rules/message_interpolation/propagated-constant.yaml

```diff
@@ -0,0 +1,11 @@
+rules:
+  - id: test_rule_id
+    patterns:
+      - pattern-inside: |
+          class $CLASS(...):
+            ...
+      - pattern: |
+            $X == $X
+    severity: INFO
+    languages: [python]
+    message: equal comparison of $X ($$X) in $CLASS
```

Comment:
You're using `$$X` here, but if I understood correctly you wanted remove that feature from this PR.